packages/create-webpack-app/src/utils/fileActions.ts

```diff
@@ -0,0 +1,221 @@
+import * as fs from "fs/promises";
+import * as ejs from "ejs";
+import { expand } from "@inquirer/prompts";
+import { spawn } from "cross-spawn";
+import * as path from "path";
+import { fileURLToPath } from "url";
+// eslint-disable-next-line node/no-missing-import
+import { logger } from "./logger.js";
+import { Answers, NodePlopAPI } from "../types";
+
+export interface GlobalConfig {
+  overwriteAll: boolean;
+}
+export interface AddConfig {
+  operationType?: string; // type of operation to be performed
+  type: string; // Type of action
+  path: string; // EJS template for file path
+  template?: string; // EJS template for file content
+  templateFile?: string; // Path to a file containing the EJS template
+  skipIfExists?: boolean; // Skip if file exists
+  transform?: (content: string, data: object) => string | Promise<string>; // Optional transform function
+  skip?: (data: object) => boolean | Promise<boolean>; // Skip function
+  force?: boolean; // Force overwrite
+  data?: Answers; // Data for EJS template rendering
+  abortOnFail?: boolean; // Abort on failure
+}
+
+const __dirname = path.dirname(fileURLToPath(import.meta.url));
+const globalConfig: GlobalConfig = { overwriteAll: false };
+const binaryExtensions = [".png", ".jpg", ".jpeg", ".gif", ".bmp", ".ico", ".pdf"];
+
+function isBinaryFile(fileName: string | undefined): boolean {
+  if (!fileName) return false;
+  const fileNameWithoutTpl = path.basename(fileName, ".tpl");
+
+  return binaryExtensions.includes(path.extname(fileNameWithoutTpl).toLowerCase());
+}
+
+async function fileExists(filePath: string): Promise<boolean> {
+  try {
+    await fs.access(filePath);
+    return true;
+  } catch {
+    return false;
+  }
+}
+
+async function compareContentWithFile(filePath: string, content: string): Promise<boolean> {
+  try {
+    const existingContent = await fs.readFile(filePath, "utf8");
+    return existingContent === content;
+  } catch {
+    return false;
+  }
+}
+
+function getDiff(filePath: string, tempFilePath: string): Promise<void> {
+  return new Promise((resolve, reject) => {
+    const editor = "code";
```

Comment:
It can be broken on some `OS`, also developer just can't have `code` package...

Reply:
initially I had vim as fallback, but then I asked on slack and changed this to vscode.
Would checking for code and then setting work?

Reply:
```typescript 
function checkIfCodeInstalled(): boolean {
  try {
    const result = spawn.sync('code', ['--version'], { stdio: 'ignore' });
    return result.status === 0;
  } catch (error) {
    return false;
  }
}

function getDiff(filePath: string, tempFilePath: string): Promise<void> {
  return new Promise((resolve, reject) => {
    const platform = process.platform;
    let editor = '';

    if (platform === 'win32') {
      if (checkIfCodeInstalled()) {
        editor = 'code';
      } else {
        return reject(new Error('Visual Studio Code is not installed. Please install VS Code to continue.'));
      }
    } else if (platform === 'darwin' || platform === 'linux') {
      editor = checkIfCodeInstalled() ? 'code' : 'vim';
    } else {
      return reject(new Error(`Unsupported platform: ${platform}`));
    }

    let diffCommand = '';

    if (editor === 'code') {
      diffCommand = `${editor} --diff ${filePath} ${tempFilePath}`;
    } else if (editor === 'vim') {
      diffCommand = `${editor} -d ${filePath ${tempFilePath}`;
    }

    const diffProcess = spawn(diffCommand, { shell: true, stdio: 'inherit' });

    diffProcess.on('exit', (code) => {
      if (code !== 0) {
        reject(new Error('Error opening diff in editor'));
      } else {
        resolve();
      }
    });
  });
}
```

Reply:
@alexander-akait Would this work?